lustrefs-exporter/benches/common/mod.rs

```diff
@@ -0,0 +1,146 @@
+};
+use injectorpp::interface::injector::*;
+use lustrefs_exporter::routes;
+use std::{
+    os::unix::process::ExitStatusExt as _,
+    process::{ExitStatus, Output},
+    time::Duration,
+};
+use tokio::{task::JoinSet, time::Instant};
+use tower::ServiceExt as _;
+
+/// Create a new Axum app with the provided state and a Request
+/// to scrape the metrics endpoint.
+fn get_app() -> (Request<Body>, Router) {
+    let app = routes::app();
+
+    let request = Request::builder()
+        .uri("/metrics?jobstats=true")
+        .method("GET")
+        .body(Body::empty())
+        .unwrap();
+
+    (request, app)
+}
+
+// Create the shared state and wrap it in an Arc so that it can be reused
+// across requests without building it each time.
+pub fn inject_mocks() {
+    let mut injector = InjectorPP::new();
+
+    injector
+        .when_called(injectorpp::func!(
+            fn(routes::jobstats_metrics_cmd)() -> Result<std::process::Child, std::io::Error>
+        ))
+        .will_execute(injectorpp::fake!(
+            func_type: fn() -> Result<std::process::Child, std::io::Error>,
+            returns: std::process::Command::new("cat")
+                .arg("../../fixtures/jobstats_only/2.14.0_162.txt")
+                .stdout(std::process::Stdio::piped())
+                .stderr(std::process::Stdio::piped())
+                .spawn()
+        ));
+
+    injector
+        .when_called_async(injectorpp::async_func!(
+            routes::lustre_metrics_output(), Result<std::process::Output, std::io::Error>
+        ))
+        .will_return_async(injectorpp::async_return!(
+            Ok(Output {
+                status: ExitStatus::from_raw(0),
+                stdout: include_str!("../../../lustre-collector/src/fixtures/valid/lustre-2.14.0_ddn133/2.14.0_ddn133_quota.txt").as_bytes().to_vec(),
+                stderr: b"".to_vec(),
+            }),
+            Result<std::process::Output, std::io::Error>
+        ));
+
+    injector
+        .when_called_async(injectorpp::async_func!(
+            routes::net_show_output(), Result<std::process::Output, std::io::Error>
+        ))
+        .will_return_async(injectorpp::async_return!(
+            Ok(Output {
+                status: ExitStatus::from_raw(0),
+                stdout: include_str!("../../fixtures/lnetctl_net_show.txt").as_bytes().to_vec(),
+                stderr: b"".to_vec(),
+            }),
+            Result<std::process::Output, std::io::Error>
+        ));
+
+    injector
+        .when_called_async(injectorpp::async_func!(
+            routes::lnet_stats_output(), Result<std::process::Output, std::io::Error>
+        ))
+        .will_return_async(injectorpp::async_return!(
+            Ok(Output {
+                status: ExitStatus::from_raw(0),
+                stdout: include_str!("../../fixtures/lnetctl_stats.txt").as_bytes().to_vec(),
+                stderr: b"".to_vec(),
+            }),
+            Result<std::process::Output, std::io::Error>
+        ));
+}
```

Comment:
I don't think we should use injectorpp here.

Reply:
I've removed this from our benchmark testing as discussed.